File: net/Functions/WeaponsController.fs
namespace Functions

open System
open System.Web.Http
open WeaponRepository

type WeaponsController () = 
    inherit ApiController()

    [<Route ("weapons")>]
    member this.getAllWeapons () = WeaponRepository.getAllWeapons ()

    [<Route ("weapons/{weaponid}/getrandomdamage")>]
    member this.getRandomDamageForWeapon weaponId = WeaponRepository.getWeapon weaponId |> Calculations.getRandomDamage

File: classesvsfunctions/ClassesAndInterfaces/WeaponsController.cs
using System;
using System.Web.Http;
using System.Collections.Generic;

namespace ClassesAndInterfaces
{
    public class WeaponsController : ApiController
    {
        IWeaponRepository weaponRepository;

        public WeaponsController (IWeaponRepository weaponRepository)
        {
            this.weaponRepository = weaponRepository;
        }

        [Route("weapons")]
        public IHttpActionResult GetAllWeapons()
        {
            IEnumerable<Weapon> allweapons = this.weaponRepository.GetAll();
            return Ok(allweapons);
        }

        [Route("weapons/{weaponid}/getrandomdamage")]
        public IHttpActionResult GetRandomDamageForWeapon(int weaponid)
        {
            return Ok(this.weaponRepository.GetWeapon(weaponid).GetRandomDamage());
        }
    }
}

